src/udp.rs

```diff
@@ -0,0 +1,34 @@
+#[cfg(unix)]
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd};
+#[cfg(unix)]
+use tokio::net::UdpSocket;
+
+#[cfg(unix)]
+// This function will be obsolete after tokio/mio internally use `socket2` and expose the methods to
+// apply options to a socket. Until then, use of `unsafe` is necessary here.
+pub fn set_buffer_sizes(
+    socket: &mut UdpSocket,
```

Comment:
I'm not seeing why this needs to have `socket` be `&mut`. The only method that is called on `socket` is `as_raw_fd`, and that takes `&self`. Did I miss something?

Reply:
On a second thought, I think you're right.

My original reasoning was: We are only able to bypass the mutability requirement because we are going through the raw file descriptor. But the mutability only refers to struct fields, not the properties that are set e.g. in the kernel.

Interesting. I would have assumed such an API needs to be designed as `mut`, but `socket2` does also not.